src/main/java/prefs/PanelInfo.java

```diff
@@ -23,4 +23,36 @@ public String getPanelFilter() {
         return this.filter;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        PanelInfo panelInfo = (PanelInfo) o;
+
+        if (name != null) {
+            if (!name.equals(panelInfo.name)) {
+                return false;
+            }
+        } else if (panelInfo.name != null) {
+            return false;
+        }
+
+        if (filter != null) {
+            if (!filter.equals(panelInfo.filter)) {
+                return false;
+            }
+        } else if (panelInfo.filter != null) {
+            return false;
+        }
```

Comment:
Sorry I didn't make this very clear last time. Is something like this better?
```java
if (name != null && !name.equals(panelInfo.name)) {
    return false;
}
if (panelInfo.name != null) {
    return false;
}

if (filter != null && !filter.equals(panelInfo.filter)) {
    return false;
}
if (panelInfo.filter != null) {
    return false;
}
```
Better still, you can use variables to make things more readable e.g.
```java
boolean isPanelNamesDifferent = name != null && !name.equals(panelInfo.name);
if (isPanelNamesDifferent) {
...
}
...
```